pkg/gameserverallocations/allocator.go

```diff
@@ -557,13 +566,16 @@ func (c *Allocator) ListenAndAllocate(ctx context.Context, updateWorkerCount int
 				req.response <- response{request: req, gs: nil, err: err}
 				continue
 			}
-			// remove the game server that has been allocated
-			list = append(list[:index], list[index+1:]...)
 
-			if err := c.allocationCache.RemoveGameServer(gs); err != nil {
-				// this seems unlikely, but lets handle it just in case
-				req.response <- response{request: req, gs: nil, err: err}
-				continue
+			if !runtime.FeatureEnabled(runtime.FeatureReuseAllocated) {
```

Comment:
...but if if you do this, the allocated game server will be out of date if you don't remove it, so you'll get _more_ conflicts rather than less.

Unless I'm missing something?

Reply:
This is the only way I found (without building out a new caching layer) to force bin packing of Counters or Lists. I.e. every single allocation request should go to the same game server. Which does result in write conflicts as the trade off.

Reply:
> Which does result in write conflicts as the trade off.

Which is why this is a really bad idea.

Reply:
I wonder if there's a way to put an Allocated game server back in the `list` after it is allocated.

It's never going to be perfect since there are multiple allocators each with their own list, but it will improve things somewhat.